src/mono/mono/metadata/object.c

```diff
@@ -931,6 +932,15 @@ compute_class_bitmap (MonoClass *klass, gsize *bitmap, int size, int offset, int
 				g_error ("compute_class_bitmap: Invalid type %x for field %s:%s\n", type->type, mono_type_get_full_name (m_field_get_parent (field)), field->name);
 				break;
 			}
+
+			// If a struct has inline array attribute, consider it as an array
```

Comment:
I am not familiar with how this GC/pointer maps work in Mono. The rest of the change looks good to me.

Reply:
A struct with the inline array attribute is handled in the same way as an array.